pkgs/by-name/tp/tplay/package.nix

```diff
@@ -0,0 +1,52 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, clang
+, ffmpeg
+, openssl
+, alsa-lib
+, libclang
+, opencv
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "tplay";
+  version = "0.4.5";
+
+  src = fetchFromGitHub {
+    owner = "maxcurzi";
+    repo = pname;
+    rev =  "v${version}";
+    hash = "sha256-qt5I5rel88NWJZ6dYLCp063PfVmGTzkUUKgF3JkhLQk=";
+  };
+
+  cargoHash = "sha256-0kHh7Wb9Dp+t2G9/Kz/3K43bQdFCl+q2Vc3W32koc2I=";
+  cargoPatches = [ ./cargo.diff ];
+  checkFlags = [
+        # requires network access
+    "--skip=pipeline::image_pipeline::tests::test_process"
+    "--skip=pipeline::image_pipeline::tests::test_to_ascii"
+    "--skip=pipeline::image_pipeline::tests::test_to_ascii_ext"
+    "--skip=pipeline::runner::tests::test_time_to_send_next_frame"
+  ];
+
+  nativeBuildInputs = [ pkg-config clang ffmpeg ];
+  buildInputs = [
+    openssl.dev
+    alsa-lib.dev
+    libclang.lib
+    ffmpeg.dev
+    opencv
+  ];
+
+  env.LIBCLANG_PATH = "${libclang.lib}/lib";
+
+  meta = {
+    description = "Terminal Media Player";
+    homepage = "https://github.com/maxcurzi/tplay";
+    platforms = lib.platforms.linux;
+    license = with lib.licenses; [ mit ];
+    maintainers = with lib.maintainers; [ demine ];
```

Comment:
```suggestion
  meta = with lib; {
    description = "Terminal Media Player";
    homepage = "https://github.com/maxcurzi/tplay";
    platforms = platforms.linux;
    license = licenses.mit;
    maintainers = with maintainers; [ demine ];
```

Reply:
A small nit. When only specifying one license you should use licenses.mit (:

Reply:
The motivation is to be found here:
[https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/meta.chapter.md](https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/meta.chapter.md)

And was primarily minded on the licenses not written in the preferred way. Then while at I suggested a general change so it was more similar to the documentation example and many other packages.

Reply:
Hm, I've also got a lot of conflicting reviews lately. Mainly because what the right approach is depends on where you read the documentation 😅

I think the point then should be that as a minimum the it should be `lib.licenses.mit` and not `with lib; [ licenses.mit ];` which I can see already has been done ☺️

Let's mark this as resolved.

Reply:
Yeah, there isn't one agreed upon policy regarding the usage of `with lib` (and other `with` statement such as `with pkgs`).

Though the _general_ sentiment seems to be that `with lib` at `meta` levels (for packages) and `options` level (for modules) are fine/very-popular, and otherwise it is down to the individual contributor (though my anecdotal experience indicates that they are _mostly_ avoided).